### ISSUE TITLE
NSubstitute.Analyzers.CSharp produces warnings about Microsoft.CodeAnalysis.CSharp in project, no analyzer output

### ISSUE THREAD

**Issue by erizzo** [nsubstitute/NSubstitute.Analyzers#77]
**Question**
I have an ordinary Test project in my solution, targeting .Net Framework 4.6.2. When I added a Nuget reference to NSubstitute.Analyzers.CSharp, I got several warnings like this:

> An instance of analyzer NSubstitute.Analyzers.CSharp.DiagnosticAnalyzers.NonVirtualSetupWhenAnalyzer cannot be created from C:\Users\erizzo\.nuget\packages\nsubstitute.analyzers.csharp\1.0.6\analyzers\dotnet\cs\NSubstitute.Analyzers.CSharp.dll : Could not load file or assembly 'Microsoft.CodeAnalysis.CSharp, Version=2.8.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified..

I then tried adding Nuget reference to Microsoft.CodeAnalysis.CSharp (I tried both 2.8.0 and the latest stable version, 2.10.0), but it made no difference.

Obviously, the NSubstitute analyzers are not producing any warnings, either.

Any ideas what I can do?

VisualStudio version is 15.6.7 (VS 2017)

**Comment by tpodolak** [comment]
@erizzo I did some digging and also asked on roslyn gitter channel, and I am afraid I don't have good news.  NSubstitute.Analyzers from the very first stable version depends on 
````c#
<PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="2.8.2" PrivateAssets="all" />
````
This is the package which provides the Roslyn goodies thanks to which we are able to analyze c# code, report diagnostic etc. When analyzer is added to project, this dependency is provided by VisualStudio itself - not by a NuGet reference/dependency. VisualStudio is shipped together with specific roslyn version, the mapping between them is listed [here](https://docs.microsoft.com/en-us/visualstudio/extensibility/roslyn-version-support?view=vs-2017). So as you can see from the link above you need Visual Studio 2017 version 15.7 or higher in order to make them work. 
At this point I can only advice to update VS to at least 15.7, would that be a problem for you @erizzo ? 
From my side I will add documentation which will clearly indicate min VS version is required for given NSubstitute.Analyzer package. I will also play around and see if it is possible to downgrade the Workspace packages to lower versions but I am affraid it might not be possible

**Comment by dtchepak** [comment]
@erizzo Thanks for taking the time to raise the issue! Logging it in the other repo isn't a problem -- we'd much prefer to have a record of the issue than have people be put off trying to decide which repo it should go in! :)

@zvirja Thanks for moving this.

@tpodolak Seems like you've already found the issue. I did find the same message mentioned in [this issue](https://github.com/dotnet/roslyn-analyzers/issues/1215), but not sure how comparable that is.
